aten/src/ATen/native/xnnpack/Allocator.cpp

```diff
@@ -0,0 +1,66 @@
+#include <ATen/native/xnnpack/Allocator.h>
+
+#ifdef USE_XNNPACK
+
+#include <ATen/native/TensorFactories.h>
+#include <c10/core/CPUAllocator.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+
+void Allocator::deleter(void* const memory) {
+  c10::free_cpu(memory);
+}
+
+DataPtr Allocator::allocate(size_t nbytes) const {
+  void* const memory = c10::alloc_cpu(nbytes + kGuard);
+
+  return DataPtr{
+      memory,
+      memory,
+      &deleter,
+      Device(DeviceType::CPU),
+  };
+}
+
+DeleterFnPtr Allocator::raw_deleter() const {
+  return deleter;
+}
+
+Tensor empty(
```

Comment:
kind of sad that it duplicates empty_cpu. Maybe just add an overload for empty_cpu that explicitly take the allocator?

Reply:
Done.

Reply:
I started integrating the allocator into empty_cpu, but that requires a change to native_functions to add c10::Allocator as a parameter, but that breaks the codegen since c10::Allocator is a not a known type there, and I assume that requires some changes in gen_pyi.py.  I am sure making that change in this PR would see some resistance so I am going to skip it.